aptos-move/aptos-aggregator/src/delta_change_set.rs

```diff
@@ -338,13 +343,35 @@ impl DeltaChangeSet {
         self,
         state_view: &impl StateView,
     ) -> anyhow::Result<WriteSet, VMStatus> {
-        let materialized_write_set = self
-            .take(state_view)
-            .expect("something terrible happened when applying aggregator deltas");
+        match self.take(state_view) {
+            Ok(materialized_write_set) => {
+                // Materialization successful, we can construct a write set.
+                WriteSetMut::new(materialized_write_set)
+                    .freeze()
+                    .map_err(|_err| {
+                        VMStatus::Error(
+                            StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR,
+                            Some("Error when freezing materialized deltas.".to_string()),
+                        )
+                    })
+            },
+            Err(vm_status) => {
+                // Error from failed delta application. Right now we do not support
```

Comment:
Why don't we actually start propagating this error? We are already making other changes to support failures in materialization, so this can be part of that effort too?

Reply:
Fair enough... let's propagate the error all the way then indeed.